pkg/operations/operations.go

```diff
@@ -93,25 +99,35 @@ func AssignedOperations() map[Operation]bool {
 
 // IsAssigned returns true when the provided operation is assigned to the pod.
 func IsAssigned(op Operation) bool {
+	operationsMtx.RLock()
+	defer operationsMtx.RUnlock()
+
 	return operations.assignedOperations[op]
 }
 
 // AssignedStringList returns a list of all operations assigned to the pod
 // as a sorted list of strings.
 func AssignedStringList() []string {
+	// Acquire a write lock since we always overwrite the existing set every time
```

Comment:
we don't overwrite the list if operations.assignedStringList != nil, so we should keep the read lock for that case so that this can remain non-blocking (list only really gets built at startup):

* acquire read lock, test for non-nil, return value if non-nil, otherwise release the read lock
* acquire write lock, build list, assign list, return list

There is a slight race condition where the list could get assigned multiple times on startup (could use sync.Once if we wanted to avoid that), but not too worried about that.

Reply:
Done.